src/emergency_stop_handling.c

```diff
@@ -0,0 +1,70 @@
+#include<stdio.h>
+#include<stdbool.h>
+#include"structures.h"
+#include"globals.h"
+#include"emergency_stop_handling.h"
+/**
+ * @file emergency_stop_handling.c
+ *
+ * @brief On account of emergency stop condition,All the elevators re moved to nearest floor.
+ * Passengers are removed from the elevators.
+ *
+ * @author Divyalakshmi Padmanaban  - dpadm049@uottawa.ca	
+ *
+ */
+void emergency_stop_handling(struct elevator ele[],int timer)
+{
+	int i,j,k=0;
+	int count;
+	bool status=true;
+	while(status)
+	{
+	
+	for(i=0;i<elevator_count;i++)
+	{   
+		    
+		    if(ele[i].passenger_count>=1)
+		    {
+		        if((ele[i].moving==1)&&(ele[i].between_floor==1))
+		        {
+			        if(ele[i].direction_up==true)
+			        {
+				        for(j=0;j<ele[i].max_passenger;j++)
+				        {
+				            ele[i].passsenger_arr[j].dest_floor=ele[i].cur_floor+1;
+				            ele[i].moving=k;
+				            ele[i].between_floor=k;
+				            remove_passenger_floor(ele[i].passenger_arr[j]);
```

Comment:
it should call passenger_drop not remove_passenger_floor

Reply:
the new parameters for passenger_drop are
`int passengers_drop(struct elevator elevator_arr[], int index, int cur_floor, unsigned int event_time);
`
elevator_arr is elevator array 
int index is the current index of the array you are using
int cur floor is the current floor of the elevator. The floor where you are dropping passengers.
unsigned int event time is the timer time the event has happened at.